core/platform/lf_zephyr_support.c

```diff
@@ -0,0 +1,525 @@
+
+    tid++; 
+
+    *thread = my_tid;   
+    return 0;
+}
+
+/**
+ * Make calling thread wait for termination of the thread.  The
+ * exit status of the thread is stored in thread_return, if thread_return
+ * is not NULL.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_thread_join(lf_thread_t thread, void** thread_return) {
+    k_thread_join(thread, K_FOREVER);
+    return 0;
+}
+
+/**
+ * Initialize a mutex.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_mutex_init(lf_mutex_t* mutex) {
+    return k_mutex_init(mutex);    
+}
+
+/**
+ * Lock a mutex.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_mutex_lock(lf_mutex_t* mutex) {
+    int res = k_mutex_lock(mutex, K_FOREVER);
+    return res;
+}
+
+/** 
+ * Unlock a mutex.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_mutex_unlock(lf_mutex_t* mutex) {
+    int res = k_mutex_unlock(mutex);
+    return res;
+}
+
+
+/** 
+ * Initialize a conditional variable.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_cond_init(lf_cond_t* cond) {
+    return k_condvar_init(cond);
+}
+
+/** 
+ * Wake up all threads waiting for condition variable cond.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_cond_broadcast(lf_cond_t* cond) {
+    k_condvar_broadcast(cond);
+    return 0;
+}
+
+/** 
+ * Wake up one thread waiting for condition variable cond.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_cond_signal(lf_cond_t* cond) {
+    return k_condvar_signal(cond);
+}
+
+/** 
+ * Wait for condition variable "cond" to be signaled or broadcast.
+ * "mutex" is assumed to be locked before.
+ * 
+ * @return 0 on success, platform-specific error number otherwise.
+ */
+int lf_cond_wait(lf_cond_t* cond, lf_mutex_t* mutex) {
+    return k_condvar_wait(cond, mutex, K_FOREVER);
+}
+
+/** 
+ * Block current thread on the condition variable until condition variable
+ * pointed by "cond" is signaled or time pointed by "absolute_time_ns" in
+ * nanoseconds is reached.
+ * 
+ * @return 0 on success, LF_TIMEOUT on timeout, and platform-specific error
+ *  number otherwise.
+ */
+int lf_cond_timedwait(lf_cond_t* cond, lf_mutex_t* mutex, instant_t absolute_time_ns) {
+    
+    // lf_sleep_until(absolute_time_ns);
+    // instant_t now;
+    // lf_clock_gettime(&now);
+    // return LF_TIMEOUT;
+    instant_t now;
+    lf_clock_gettime(&now);
+    interval_t sleep_duration_ns = absolute_time_ns - now;
+    k_timeout_t timeout = K_NSEC(sleep_duration_ns);
+    int res = k_condvar_wait(cond, mutex, timeout);
+    if (res == 0) {
+        return 0;
+    } else {
+        return LF_TIMEOUT;
+    }
+}
+
+// Atomics
+//  Implemented by just entering a critical section and doing the arithmetic
+//  FIXME: We are now restricted to atomic integer operations
```

Comment:
What does this FIXME mean? Do we need anything more than atomic integer operations?

Also, these functions need documentation.

Reply:
Removed it and added documentation